pkg/iss/iss.go

```diff
@@ -424,6 +424,11 @@ func New(
 		}
 		chkp := checkpoint.StableCheckpointFromPb(_chkp.Pb())
 
+		if err := chkp.SyntacticCheck(params); err != nil {
```

Comment:
This is included in the verification of the checkpoint that is done later. I understand that we need that syntactic check first, since we access some fields before the full verification, but maybe we could still just run the full verification first (i.e., here). It is less efficient (especially if a malicious node makes us verify plenty of signatures uselessly - but it can do that anyway), but it's much cleaner to read and think about.

Reply:
Yes, you got my train of thought when I called it there. Let us do it as you say despite the slight potential inefficiency (We can always put the check of whether the node is part of the membership higher up with an extra check or so). Done.

Reply:
I have readded the syntactic check to after the checkpoint is reconstructed from Pb, where it used to be before :). I also readded the simpler check of the node being part of the membership earlier again as an optimzation.

Reply:
Perfect. In fact, checking if the node is in the checkpoint's membership is not an optimization (I didn't realize you had removed that too). The checkpoint can be perfectly valid and still not include the receiving node in its membership, so this check cannot go inside the checkpoint's `Verify` method. But in ISS we still need to check it, because if we tried to restore state from a valid checkpoint where we are not a member, things would break.
You can resolve this convo after reading this, the code is good now as it is.